adversarial_workflow/utils/citations.py

```
@@ -0,0 +1,608 @@
+    offset = 0  # Track offset as we insert badges
+
+    for match in URL_PATTERN.finditer(document):
+        url = match.group().rstrip(".,;:")
+        if url in url_results:
+            result = url_results[url]
+            badge = get_status_badge(result)
+
+            # Check if badge already exists after this URL
+            end_pos = match.end() + offset
+            remaining = marked[end_pos:]
+            if remaining.startswith(" [✅") or remaining.startswith(" [⚠️") or remaining.startswith(" [❌") or remaining.startswith(" [🔄"):
```

Ruff (0.14.14)

[warning] Call `startswith` once with a `tuple`

Merge into a single `startswith` call

(PIE810)